cosmos/models/Workflow.py

```diff
@@ -243,6 +249,7 @@ def params_or_signature_default_or(name, default):
                         mem_req=mem_req if mem_req is not None else params_or_signature_default_or('mem_req', None),
                         time_req=time_req,
                         successful=False,
+                        max_attempts=max_attempts if max_attempts is not None else params_or_signature_default_or('max_attempts', None),
```

Comment:
should we not check params for this, and only have one way to set it (in add_task)?

Reply:
Yeah, I was blindly copying mem_req's behavior; I can change this.

Reply:
Resolved in 6cb0562